designsafe/apps/api/datafiles/operations/agave_operations.py

```diff
@@ -0,0 +1,587 @@
+        _ext = os.path.splitext(file_name)[1].lower()
+        _name = os.path.splitext(file_name)[0]
+        now = datetime.datetime.utcnow().strftime('%Y-%m-%d %H-%M-%S')
+        trash_name = '{}_{}{}'.format(_name, now, _ext)
+    except HTTPError as err:
+        if err.response.status_code != 404:
+            logger.error("Unexpected exception listing path {} under .trash in system {}".format(file_name, system))
+            raise
+
+    resp = move(client, system, path, system, trash_path, trash_name)
+
+    return resp
+
+
+def upload(client, system, path, uploaded_file, webkit_relative_path=None, *args, **kwargs,):
+    """Upload a file.
+    Params
+    ------
+    client: agavepy.agave.Agave
+        Tapis client to use.
+    system: str
+        Tapis system ID for the file.
+    path: str
+        Path to upload the file to.
+    uploaded_file: file
+        File object to upload.
+
+    Returns
+    -------
+    dict
+    """
+
+    # try:
+    #    listing(client, system=system, path=path)
+    # except HTTPError as err:
+    #    if err.response.status_code != 404:
+    #        raise
+    if webkit_relative_path:
+        rel_path = os.path.dirname(webkit_relative_path).strip('/')
+        mkdir(client, system, path, rel_path)
+        path = os.path.join(path, rel_path)
+
+    upload_name = os.path.basename(uploaded_file.name)
+
+    resp = client.files.importData(systemId=system,
+                                   filePath=urllib.parse.quote(path),
+                                   fileName=str(upload_name),
+                                   fileToUpload=uploaded_file)
+
+    agave_indexer.apply_async(kwargs={'systemId': system,
+                                      'filePath': path,
+                                      'recurse': False},
+                              queue='indexing'
+                              )
+
+    return dict(resp)
+
+
+def preview(client, system, path, href, max_uses=3, lifetime=600):
+    """Preview a file.
+    Params
+    ------
+    client: agavepy.agave.Agave
+        Tapis client to use.
+    system: str
+        Tapis system ID.
+    path: str
+        Path to the file.
+    href: str
+        Tapis href for the file to be previewed.
+    max_uses: int
+         Maximum amount the postit link can be used.
+    lifetime: int
+        Life time of the postit link in seconds.
+
+    Returns
+    -------
+    dict
+    """
+
+    file_name = path.strip('/').split('/')[-1]
+    file_ext = os.path.splitext(file_name)[1].lower()
+
+    if not href:
+        href = client.files.list(systemId=system, filePath=path)[0]['_links']['self']['href']
+
+    args = {
+        'url': urllib.parse.unquote(href),
+        'maxUses': max_uses,
+        'method': 'GET',
+        'lifetime': lifetime,
+        'noauth': False
+    }
+
+    result = client.postits.create(body=args)
+    url = result['_links']['self']['href']
+
+    if file_ext in settings.SUPPORTED_TEXT_PREVIEW_EXTS:
```

Comment:
the determination of the `file_ext` is repeated in each different `_operation.py` file.  consider refactoring.